server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -640,7 +640,7 @@ public void updateShardState(
                     if (currentRouting.initializing() && currentRouting.isRelocationTarget() == false && newRouting.active()) {
                         // the cluster-manager started a recovering primary, activate primary mode.
                         replicationTracker.activatePrimaryMode(getLocalCheckpoint());
-                        ensurePeerRecoveryRetentionLeasesExist();
+                        postActivatePrimaryMode();
```

Comment:
nit - should we move this to within the activatePrimaryMode method. We can pass the ` postActivatePrimaryMode()` runnable during ReplicationTracker ctor. WDYT?

Reply:
Makes sense. Currently, I have combined the logic with `ensurePeerRecoveryRetentionLeasesExist`. Created an issue to track the suggested change: https://github.com/opensearch-project/OpenSearch/issues/10863

Reply:
The logic is spread too sparsely and might take a maintainability hit. Why can't we do it as a part of activating primary mode itself?